bench/workgen/runner/example_dynamic_tables.py

```diff
@@ -27,139 +27,54 @@
-workload.options.run_time = 10
-
-# Start the workload.
-workload_thread = ThreadWithReturnValue(target=workload.run, args=([connection]))
-workload_thread.start()
-
-# Create and drop tables while the workload is running.
-tables_to_delete = []
-while workload_thread.is_alive():
-    create(session, workload, table_config)
-
-    # Select a random table to drop.
-    if len(tables):
-        idx = random.randint(0, len(tables) - 1)
-        uri = tables[idx]
-        # Mark it for deletion.
-        drop(workload, uri)
-
-    # Trigger the Workgen garbage collection.
-    tables_to_delete += workload.garbage_collection()
-    deleted_tables = []
-    for t in tables_to_delete:
-        try:
-            session.drop(t)
-            # If the WiredTiger call is successful, this table has been removed safely from all
-            # layers.
-            deleted_tables.append(t)
-        except WiredTigerError as e:
-            assert "device or resource busy" in str(e).lower()
-
-    # Update the list of tables pending deletion.
-    for t in deleted_tables:
-        tables_to_delete.remove(t)
-
-    time.sleep(1)
-
-assert workload_thread.join() == 0
-
-# It is possible that Python and Workgen are not matching at this point in terms of tables. Since
-# Workgen cannot remove a table that is still being used, it may have kept reference to it.
-# Now the workload is finished, call the garbage collector on Workgen, nothing should be blocking
-# the deletion of tables that are supposed to be removed.
-# Note that we are not checking what's on disk, we don't need to use what garbage_collection
-# returns.
-workload.garbage_collection()
-
-# Check tables match between Python and Workgen.
-workgen_tables = workload.get_tables()
-
-assert len(tables) == len(workgen_tables)
-for t in tables:
-    assert t in workgen_tables
+thread_upd_rnd = Thread(op_upd_rnd * 5)
+thread_read_rnd = Thread(op_read_rnd * 5)
+
+workload = Workload(context, thread_ins_rnd + thread_upd_rnd + thread_read_rnd)
+workload.options.run_time = 300
+
+# Start a workload thread to create tables periodically:
+# Create 3 tables every 5 seconds whenever the database size falls below 100 MB.
+# Stop creating the tables once the database size reached 200 MB.
+workload.options.dynamic_create_period = 5
+workload.options.dynamic_create_count = 3
+workload.options.dynamic_create_trigger = 100
+workload.options.dynamic_create_target = 200
+
+# Start a workload thread to drop tables periodically:
+# Drop 2 tables every 3 seconds whenever the database size goes above 250 MB.
+# Stop dropping the tables once the database size reaches 50 MB.
+workload.options.dynamic_drop_period = 3
+workload.options.dynamic_drop_count = 2
+workload.options.dynamic_drop_trigger = 250
+workload.options.dynamic_drop_target = 50
+
+ret = workload.run(connection)
+assert ret == 0, ret
+
+end_time = time.time()
+run_time = end_time - start_time
+
+print('Workload took %d seconds' %(run_time))
```

Comment:
Is it necessary while we have a `run_time` option?

Reply:
No, not really - I was testing something - removed it.